pallet/src/signer.rs

```diff
@@ -51,25 +56,37 @@ pub struct SignerData {
     pub cheque_limit: u128,
 }
 
+/// Storage struct definition for a multi-signer request.
 #[derive(Clone, Eq, PartialEq, Encode, Decode, RuntimeDebug, TypeInfo, MaxEncodedLen)]
 #[scale_info(skip_type_params(Size))]
-pub struct Multisig<AccountId, Size>(BoundedBTreeMap<AccountId, SignerData, Size>)
+pub struct Multisig<AccountId, BlockNumber, Size>
 where
     AccountId: Parameter + Ord + MaybeSerializeDeserialize,
-    Size: Get<u32>;
+    BlockNumber: Parameter + Ord + MaybeSerializeDeserialize + Default,
+    Size: Get<u32>,
+{
+    /// The signers of an script transaction.
+    signers: BoundedBTreeMap<AccountId, SignerData, Size>,
+    /// The block number when this `Multisig` was created and stored.
+    block_num: BlockNumber,
```

Comment:
possible alternative
```suggestion
    block_height: BlockNumber,
```

Reply:
I also like `block_height`, and this term is used frequently, but it is still strange that one core method is called `frame_system::Pallet::block_number()`.

Reply:
I updated the variable name but not the method. Shall we change this too?